### PR TITLE
feat(legacy): Improved RoundedRect & GradientShader FPS Peformance

### PR DESCRIPTION
**Pull request by EclipsesDev**
- Optimized rendering algorithms for RoundedRect and GradientShader
- Reduced redundant calculations for rendering.

Now FPS should be significantly more higher/stable.

### CODE REVIEW

File: src/main/java/net/ccbluex/liquidbounce/utils/render/RenderUtils.kt [hunk 1]
```diff
@@ -358,34 +358,34 @@ object RenderUtils : MinecraftInstance() {
...
-                val x = cx + radiusF * sin(angle)
-                val y = cy + radiusF * cos(angle)
-                worldrenderer.pos(x, y, 0.0).color(red, green, blue, alpha).endVertex()
+            for (i in 0..90 step 10) {
+                val angle = Math.toRadians(startAngle + i)
+                val x = cx + radiusD * sin(angle)
+                val y = cy + radiusD * cos(angle)
+                glVertex2d(x, y)
             }
         }
```
Comment (superblaubeere27) [inline src/main/java/net/ccbluex/liquidbounce/utils/render/RenderUtils.kt:380]: Why is this an performance improvement?

File: src/main/java/net/ccbluex/liquidbounce/utils/render/RenderUtils.kt [hunk 2]
```diff
@@ -512,42 +512,40 @@ object RenderUtils : MinecraftInstance() {
...
 
-        tessellator.draw()
+        glEnd()
```
Comment (superblaubeere27) [inline src/main/java/net/ccbluex/liquidbounce/utils/render/RenderUtils.kt:542]: s.a.

**Review by superblaubeere27 — COMMENTED** [review]
Why does this improve performance? Did you measure it?

File: src/main/java/net/ccbluex/liquidbounce/utils/render/RenderUtils.kt (hunk 1 above)
Comment (EclipsesDev) [reply src/main/java/net/ccbluex/liquidbounce/utils/render/RenderUtils.kt:380]: Not sure but using tesselator in my testing get less fps, well I'm assuming tesselator is better to be used in 3d then 2d rendering

File: src/main/java/net/ccbluex/liquidbounce/utils/render/RenderUtils.kt (hunk 2 above)
Comment (EclipsesDev) [reply src/main/java/net/ccbluex/liquidbounce/utils/render/RenderUtils.kt:542]: the higher the step, the less smooth the rounded edges is, meaning less lag.. in my testing step 90 made the edges straight, (so the rect is like a hexagon)

**Pull request merged**